src/Aspire.Dashboard/Components/Dialogs/SettingsDialog.razor.cs

```diff
@@ -44,6 +70,21 @@ private async Task SettingChangedAsync()
         }
     }
 
+    private void OnLanguageChanged()
+    {
+        if (_selectedUiCulture is null || StringComparers.Culture.Equals(CultureInfo.CurrentUICulture.Name, _selectedUiCulture.Name))
+        {
+            return;
+        }
+
+        var uri = new Uri(NavigationManager.Uri)
+            .GetComponents(UriComponents.PathAndQuery, UriFormat.Unescaped);
+
+        NavigationManager.NavigateTo(
+            DashboardUrls.SetLanguageUrl(_selectedUiCulture.Name, uri),
+            forceLoad: true);
```

Comment:
Setting the cookie isn't the problem. Updating text and loading the new satellite assemblies is the real issue.

> Did you consider setting the cookie in the browser with JS interop like changing the theme? That plus triggering a StateHasChanged on the overall page to rerender everything seems like it could work. Although I wouldn't be surprised if changing the culture in the middle of a HTTP request is problematic.

I did, but this approach doesn't work. I've read more into this the past day, and while some workarounds are possible (see [here](https://github.com/dotnet/runtime/issues/98415)), I would strongly prefer continuing with the reload strategy for the following reasons:
- calling the undocumented internal api is not compatible with our current blazor setup
- we could download all resources, but that would require downloading more, with limited use
- changing the language is probably less likely than changing the theme

Reply:
Yes, there are no workarounds for what we are doing. Those are the only workarounds I have found

Reply:
I'll take a look and see whether I can find a way to make it work.

If a refresh is required then I think a confirmation should be displayed after changing the language, e.g.

> The dashboard must be refreshed when the language is changed. Are you sure you want to change the language?
> Yes, No

Reply:
> If a refresh is required then I think a confirmation should be displayed after changing the language, e.g.

I don’t think this would be very useful. The user is pretty clear in their intent, the dialog adds unnecessary friction. Reloading (or opening a new page to select language in) is a common pattern, including on Microsoft websites

Reply:
But it’s inconsistent with the rest of the website. Nothing else (adding filters, changing the theme, etc) triggers the dashboard to completely reload. Some page state could be lost.

Reply:
Yeah, I think it would look better below. And slightly smaller text.

Reply:
![image](https://github.com/user-attachments/assets/2785aad6-8913-4777-b93d-bac92a6c452d)

Reply:
What do you think of this?